src/app/task-details/task-details.component.ts

```diff
@@ -26,28 +31,73 @@ export class TaskDetailsComponent implements OnInit {
     });
   }
 
+
   ngOnInit(): void {
-    const taskId = +this.route.snapshot.paramMap.get('id')!;
-    this.task = this.taskService.getTasks()[taskId];
-    if (this.task){
-      this.taskForm.setValue({
-        title: this.task.title,
-        description: this.task.description,
-        type: this.task.type,
-        status: this.task.status
+    this.route.paramMap
+      .pipe(
+        takeUntil(this.destroy$),
+        switchMap(paramMap => {
+          const taskID = paramMap.get('id');
```

Comment:
you should have a taskId otherwise you can't go to task-details page (assuming the router has a path variable id)